File: rental-backend/src/main/java/com/homeflex/features/vehicle/dto/response/VehicleBookingResponse.java
package com.homeflex.features.vehicle.dto.response;

import com.homeflex.features.vehicle.domain.enums.VehicleBookingStatus;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record VehicleBookingResponse(
        UUID id,
        UUID vehicleId,
        VehicleResponse vehicle,
        UUID tenantId,
        com.homeflex.core.dto.response.UserDto tenant,
        LocalDate startDate,
        LocalDate endDate,
        BigDecimal totalPrice,
        String currency,
        VehicleBookingStatus status,
        BigDecimal platformFee,
        String message,
        String rejectionReason,
        String stripePaymentIntentId,
        LocalDateTime createdAt
) {}

File: rental-app-flutter/lib/features/vehicles/models/vehicle_booking_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'vehicle_booking_response.freezed.dart';
part 'vehicle_booking_response.g.dart';

@freezed
abstract class VehicleBookingResponse with _$VehicleBookingResponse {
  const factory VehicleBookingResponse({
    required String id,
    required String vehicleId,
    required String tenantId,
    required String startDate,
    required String endDate,
    double? totalPrice,
    required String currency,
    required String status,
    double? platformFee,
    String? message,
    required String createdAt,
  }) = _VehicleBookingResponse;

  factory VehicleBookingResponse.fromJson(Map<String, dynamic> json) =>
      _$VehicleBookingResponseFromJson(json);
}
